mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -462,36 +462,55 @@ final class OnlineLDAOptimizer extends LDAOptimizer {
     val expElogbetaBc = batch.sparkContext.broadcast(expElogbeta)
     val alpha = this.alpha.asBreeze
     val gammaShape = this.gammaShape
+    val optimizeDocConcentration = this.optimizeDocConcentration
+    // We calculate logphat in the same pass as other statistics, but we only need
+    // it if we are optimizing docConcentration
+    val logphatPartOptionBase = () => if (optimizeDocConcentration) Some(BDV.zeros[Double](k))
+                                      else None
 
-    val stats: RDD[(BDM[Double], List[BDV[Double]])] = batch.mapPartitions { docs =>
+    val stats: RDD[(BDM[Double], Option[BDV[Double]], Int)] = batch.mapPartitions { docs =>
```

Comment:
Let's use Long for the doc count since it could overflow for large datasets and miniBatchFraction

Reply:
Done.